src/processor/reconnection-graph.service.ts

```diff
@@ -431,12 +429,47 @@ export class ReconnectionGraphService {
   }
 
   async processChainEvents(promises: Promise<ParsedEventResult>[], dsnpUserId: MessageSourceId): Promise<void> {
-    // loop over promises and wait for all to resolve
-    for (const promise of promises) {
-      const [event, eventMap] = await promise;
-      if (!event) {
-        throw new Error(`Error submitting extrinsic`);
+    // iterate over promises and wait for all to resolve
+    try {
+      await Promise.all(promises);
+      this.logger.debug(`Successfully processed chain events for ${dsnpUserId.toString()}`);
+
+      // iterate over promises and check for `BatchCompleted` event
+      for (const promise of promises) {
+        const [event, eventMap] = await promise;
+        if (!event) {
+          // if we dont get any events, covering any unexpected connection errors
+          throw new Error(`No events were found for ${dsnpUserId.toString()}`);
+        }
+        if(this.blockchainService.api.events.utility.BatchCompleted.is(event)) {
+          await this.process_batch_completed_event(event, eventMap, dsnpUserId);
+        } else {
+          if(this.blockchainService.api.events.utility.BatchInterrupted.is(event)) {
+            this.logger.log(`BatchInterrupted event found for ${dsnpUserId.toString()}`);
+            await this.process_batch_interrupted_event(event, eventMap, dsnpUserId);
+          } else {
+            this.logger.warn(`Unexpected event found for ${dsnpUserId.toString()}`);
+            this.logger.warn(event);
+            this.logger.warn(eventMap);
+          }
```

Comment:
note: since we run signed extension checks, we wont expect `BatchInterrupted` event unless our actual configs, schemaids etc are wrong.